tests/Scout/ScoutEngineTest.php

```diff
@@ -0,0 +1,548 @@
+            ],
+        ];
+
+        yield 'object' => [
+            [
+                (object) ['_id' => 'key_1', 'foo' => 'bar'],
+                (object) ['_id' => 'key_2', 'foo' => 'bar'],
+            ],
+        ];
+
+        yield Document::class => [
+            [
+                Document::fromPHP(['_id' => 'key_1', 'foo' => 'bar']),
+                Document::fromPHP(['_id' => 'key_2', 'foo' => 'bar']),
+            ],
+        ];
+
+        yield BSONDocument::class => [
+            [
+                new BSONDocument(['_id' => 'key_1', 'foo' => 'bar']),
+                new BSONDocument(['_id' => 'key_2', 'foo' => 'bar']),
+            ],
+        ];
+    }
+
+    public function testUpdate(): void
+    {
+        $date = new DateTimeImmutable('2000-01-02 03:04:05');
+        $database = m::mock(Database::class);
+        $collection = m::mock(Collection::class);
+        $database->shouldReceive('selectCollection')
+            ->with('collection_indexable')
+            ->andReturn($collection);
+        $collection->shouldReceive('bulkWrite')
+            ->once()
+            ->with([
+                [
+                    'updateOne' => [
+                        ['_id' => 'key_1'],
+                        ['$setOnInsert' => ['_id' => 'key_1'], '$set' => ['id' => 1, 'date' => new UTCDateTime($date)]],
+                        ['upsert' => true],
+                    ],
+                ],
+                [
+                    'updateOne' => [
+                        ['_id' => 'key_2'],
+                        ['$setOnInsert' => ['_id' => 'key_2'], '$set' => ['id' => 2]],
+                        ['upsert' => true],
+                    ],
+                ],
+            ]);
+
+        $engine = new ScoutEngine($database, softDelete: false);
+        $engine->update(EloquentCollection::make([
+            new SearchableModel([
+                'id' => 1,
+                'date' => $date,
+            ]),
+            new SearchableModel([
+                'id' => 2,
+            ]),
+        ]));
+    }
+
+    public function testUpdateWithSoftDelete(): void
+    {
+        $date = new DateTimeImmutable('2000-01-02 03:04:05');
+        $database = m::mock(Database::class);
+        $collection = m::mock(Collection::class);
+        $database->shouldReceive('selectCollection')
+            ->with('collection_indexable')
+            ->andReturn($collection);
+        $collection->shouldReceive('bulkWrite')
+            ->once()
+            ->with([
+                [
+                    'updateOne' => [
+                        ['_id' => 'key_1'],
+                        ['$setOnInsert' => ['_id' => 'key_1'], '$set' => ['id' => 1]],
+                        ['upsert' => true],
+                    ],
+                ],
+            ]);
+
+        $model = new SearchableModel(['id' => 1]);
+        $model->delete();
+
+        $engine = new ScoutEngine($database, softDelete: false);
```

Comment:
The test case is "testUpdateWithSoftDelete" but you're specifying `softDelete: false` here. Is that a mistake?

Reply:
Good catch, I updated the test.